lib/OpenCloud/LoadBalancer/Service.php

```diff
@@ -51,11 +52,20 @@ public function loadBalancer($id = null)
      */
     public function loadBalancerList($detail = true, array $filter = array())
     {
+        $options = $this->makeResourceIteratorOptions($this->resolveResourceClass('LoadBalancer'));
+
+        if (isset($filter['limit'])) {
+            $options['limit.page'] = $filter['limit'];
```

Comment:
Could you add a comment here explaining why this the next 2 statements are necessary? I'm particularly curious about the `unset`.